nixos/modules/services/hardware/acpid.nix

```diff
@@ -3,153 +3,189 @@
+
+              ::: {.note}
+              Many events will require kernel modules to be available via `boot.initrd.availableKernelModules`.
+              :::
+            '');
+
+            example = literalExpression ''
+              "button/power.*" "button/lid.*" "ac_adapter.*" "button/mute.*" "button/volumedown.*" "cd/play.*" "cd/next.*"
             '';
           };
-        };
-      };
-
-      powerEventCommands = mkOption {
-        type = types.lines;
-        default = "";
-        description = lib.mdDoc "Shell commands to execute on a button/power.* event.";
-      };
 
-      lidEventCommands = mkOption {
-        type = types.lines;
-        default = "";
-        description = lib.mdDoc "Shell commands to execute on a button/lid.* event.";
+          action = mkOption {
+            type = types.lines;
+            description = "Shell commands to execute when the event is triggered";
+          };
+        };
+      });
+
+      description = lib.mdDoc ''
+        Event handlers
+
+        ::: {.note}
+        Handler can be a single command.
+        :::
+      '';
+
+      default = {};
+      example = {
+        ac-power = {
+          event = "ac_adapter/*";
+
+          action = ''
+            vals=($1)  # space separated string to array of multiple values
+            case ''${vals[3]} in
+              00000000)
+                echo unplugged >> /tmp/acpi.log
+                ;;
+              00000001)
+                echo plugged in >> /tmp/acpi.log
+                ;;
+              *)
+                echo unknown >> /tmp/acpi.log
+                ;;
+            esac
+          '';
+        };
       };
+    };
 
-      acEventCommands = mkOption {
-        type = types.lines;
-        default = "";
-        description = lib.mdDoc "Shell commands to execute on an ac_adapter.* event.";
-      };
+    powerEventCommands = mkOption {
+      type = types.lines;
+      default = "";
+      description = "Shell commands to execute on a button/power.* event";
+    };
 
+    lidEventCommands = mkOption {
+      type = types.lines;
+      default = "";
+      description = "Shell commands to execute on a button/lid.* event";
     };
 
+    acEventCommands = mkOption {
+      type = types.lines;
+      default = "";
+      description = "Shell commands to execute on an ac_adapter.* event";
+    };
   };
 
-
-  ###### implementation
-
-  config = mkIf cfg.enable {
-
+  mkConfig = cfg: mkIf cfg.enable {
     systemd.services.acpid = {
       description = "ACPI Daemon";
       documentation = [ "man:acpid(8)" ];
 
-      wantedBy = [ "multi-user.target" ];
+      # Acpid requires /var/run to exist but dosen't create it, and no systemd service seems to create it either.
+      # Note this is only a problem for stage1.
+      script = ''
+        ln -s /run /var/run || true
+
+        ${cfg.package}/bin/acpid \
+          --foreground \
+          --netlink \
+          --confdir "${acpiConfFor cfg}/handlers" \
```

Comment:
Isn't the trailing \ causing problems when logEvents is not appended?

Reply:
No, it behaves similarly to running it interactively
```sh
> echo hi \
>  \
>  \
>  \
>
hi
>
```